From dbc616d6a408d3e7c801ce8de4f86652456ed7af Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Mon, 12 Apr 2021 12:18:26 +0300
Subject: [PATCH 32/42] clk: scmi: Correct return of set_rate callback

set_rate callback of strunct set_rate requires as return
value the frequency that was set or a negative value in case
of a failure.

Upstream-Status: Pending

Issue: ALB-6968
Signed-off-by: Ghennadi Procopciuc <Ghennadi.Procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/clk/clk_scmi.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/clk/clk_scmi.c b/drivers/clk/clk_scmi.c
index 49a7656dec..767196bbf4 100644
--- a/drivers/clk/clk_scmi.c
+++ b/drivers/clk/clk_scmi.c
@@ -1,6 +1,7 @@
 // SPDX-License-Identifier: GPL-2.0+
 /*
  * Copyright (C) 2019-2020 Linaro Limited
+ * Copyright 2021 NXP
  */
 #include <common.h>
 #include <clk-uclass.h>
@@ -135,7 +136,10 @@ static ulong scmi_clk_set_rate(struct clk *clk, ulong rate)
 	if (rc)
 		return 0;
 
-	return scmi_to_linux_errno(out.status);
+	if (scmi_to_linux_errno(out.status))
+		return 0;
+
+	return rate;
 }
 
 static const struct clk_ops scmi_clk_ops = {
-- 
2.25.1

